lib/hccrawler.js

```diff
@@ -322,49 +321,41 @@ class HCCrawler extends EventEmitter {
    * @return {!Promise<!Array<!string>>}
    * @private
    */
-  _request(options, depth, retryCount = 0) {
+  async _request(options, depth, retryCount = 0) {
     this.emit(HCCrawler.Events.RequestStarted, options);
-    return this._newPage(options)
-      .then(crawler => {
-        this.emit(HCCrawler.Events.NewPage, crawler.page());
-        return crawler.crawl()
-          .then(res => {
-            res = extend({ options, depth }, res);
-            this.emit(HCCrawler.Events.RequestFinished, res);
-            return this._success(res)
-              .then(() => void this._exportLine(res))
-              .then(() => crawler.close())
-              .then(() => res.links);
-          })
-          .catch(error => {
-            if (retryCount >= options.retryCount) throw error;
-            this.emit(HCCrawler.Events.RequestRetried, options);
-            return this._removeExists(options)
```

Comment:
it doesn't have to remove `exists` mark because retry doesn't go though `_checkRequested` method